collect_app/src/test/java/org/odk/collect/android/preferences/JsonPreferencesGeneratorTest.kt

```diff
@@ -66,4 +67,63 @@ class JsonPreferencesGeneratorTest : TestCase() {
         assertThat(jsonPrefs, not(containsString("password")))
         assertThat(jsonPrefs, not(containsString("123456")))
     }
+
+    @Test
+    fun `Only saved settings should be included in json`() {
```

Comment:
Not to suggest they're not useful, but I'm interested in why you decided to write these tests?

Reply:
As we discussed during our meeting those tests are for the fix from https://github.com/getodk/collect/pull/4587/commits/7a22d846f9309612b9c54ff5ffe5a2c560fa0694